packages/lib/next-auth/auth-options.ts

```diff
@@ -90,6 +92,26 @@ export const NEXT_AUTH_OPTIONS: AuthOptions = {
           }
         }
 
+        if (!user.emailVerified) {
+          const [lastUserVerificationToken] = await getLastVerificationToken({ userId: user.id });
+
+          if (!lastUserVerificationToken) {
+            await sendConfirmationToken({ email });
+            throw new Error(ErrorCode.UNVERIFIED_EMAIL);
+          }
+
```

Comment:
should the `expiredToken`, `lastSentToken` and `sentWithinLastHour` be extracted into a separate function? or is it fine like this?